### ISSUE TITLE
Colored (RGBA) EXR images  generated from Godot have wrong order of channels. 

### ISSUE THREAD

**Issue by leonard7e** [godotengine/godot#55472]
### Godot version

3.4 stable

### System information

Ubuntu 20.04

### Issue description

EXR images generated from Godot changes channel order. Resulting in distorted colors, if the exr file is opened with other apps. Not limited to Blender and Gimp, but includes other apps like Da Vinci resolve.

Images created from Godot will show distorted colors in exrdisplay.

### Steps to reproduce

You do need an external EXR viewer for comparison.

Best, use exrdisplay, which comes from makers of OpenEXR. It is shipped with the OpenEXR bundle. 
https://www.openexr.com/using.html

The steps:
run the Godot project shipped here. It will load an RGB image. Hit the "Generate EXR" button. It will write the image to file "generated_color_testing.exr"

Now, open "generated_color_testing.exr" in exrdisplay (or DaVinciResolve, Blender, and so on). Compare "generated_color_testing.exr" with "color_testing_reference.exr" (which shows correct coloring in exrdisplay).

### Minimal reproduction project

[minimal_exr_godot.zip](https://github.com/godotengine/godot/files/7626365/minimal_exr_godot.zip)

**Comment by fire** [comment]
Investigating

**Comment by fire** [comment]
@leonard7e @RodZill4 Can you test?

https://github.com/godotengine/godot/pull/56715

**Comment by leonard7e** [comment]
> @leonard7e @RodZill4 Can you test?

@fire 

I tried. I only found Mono builds for Linux.  (I don't have experiences with Mono on Linux). 
I used  the linux-editor-mono artifact from 
 https://github.com/godotengine/godot/actions/runs/1686602957
Godot / Mono crashed.

```
Godot Engine v3.5.beta.mono.custom_build.cc55cc459 - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce RTX 2060 SUPER/PCIe/SSE2
Async. shader compilation: OFF
/lib/x86_64-linux-gnu/libasound.so.2: undefined symbol: snd_dlpath
 
Mono: Log file is: '/home/leonard/.local/share/godot/mono/mono_logs/2022-01-12_10.29.30_64834.log'
ERROR: Mono: FATAL ERROR 'Runtime critical type Mono.RuntimeStructs/MonoError not found (in domain , error)', ABORTING! Logfile: '/home/leonard/.local/share/godot/mono/mono_logs/2022-01-12_10.29.30_64834.log'.
   at: mono_log_callback (modules/mono/mono_gd/gd_mono_log.cpp:91)

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.

…
… 

fish: “./godot.x11.opt.tools.64.mono” terminated by signal SIGABRT (Abort)

```

**Comment by fire** [comment]
Mono Linux builds have been crashing for me for several months now.

**Comment by leonard7e** [comment]
Mh … I compiled Godot from your branch/repo. It worked. The colors look correct.
![Screenshot_2022-01-12_11-19-46](https://user-images.githubusercontent.com/5395270/149121937-abb1d77f-e14c-403a-a23b-27eb1fe08ba5.png)

There is a curious indent, where color shapes overlap

**Comment by fire** [comment]
Fixed by: https://github.com/godotengine/godot/pull/56715

**Comment by fire** [comment]
@leonard7e If you can you provide more details on this. I have no idea how to fix that indent, but the color channels was obvious :D

**Comment by leonard7e** [comment]
@fire 
I tried to reproduce the indent issue with different EXR files. But I was not able to do.  
(EXR rgb and rgba. With/without transparent background.)
Results seems correct.  I am puzzled

**Comment by danilw** [comment]
just tested on **Godot_v3.4.2-stable** still same problem, EXR generated by Godot has wrong channel order when I open it in Blender or GIMP (on screenshot original exr saved and opened in godot (bottom image in Godot(top left), GIMP(bot left), blender(right))

![1](https://user-images.githubusercontent.com/24825887/151652240-e1e65872-9d47-4875-8a46-a40e2a87feaf.png)

**This test Godot project** with exr included - [test_exr_save.zip](https://github.com/godotengine/godot/files/7963194/test_exr_save.zip)

exr generated by **Godot** _SaveExr_ has **BGRA** channel order

This **example file** https://github.com/AcademySoftwareFoundation/openexr-images/blob/master/v2/LowResLeftView/composited.exr has **ABGR** channels order

_test_tinyexr_ from https://github.com/syoyo/tinyexr if use EXR generated by Godot, _test_tinyexr_ save it with **ABGR** channels order (and it will be opened correctly in GIMP/Blender)

**Comment by fire** [comment]
I am not sure if it is in 3.4, but the tagging says 3.5. Good night.

**Comment by danilw** [comment]
look like it fixed in 3.5 https://github.com/godotengine/godot/pull/56715 so nwm, sorry for up
